pkg/operator/ceph/csi/spec.go

```diff
@@ -294,6 +296,11 @@ func (r *ReconcileCSI) startDrivers(ver *version.Info, ownerInfo *k8sutil.OwnerI
 		tp.EnableOMAPGenerator = true
 	}
 
+	// if k8s >= v1.20 enable oidc token projection.
+	if ver.Major == KubeMinMajor && ver.Minor >= KubeMinVerForOIDCTokenProjection {
+		tp.EnableOIDCTokenProjection = true
+	}
```

Comment:
As this feature is backed by feature-gate if someone is still on the older kubernetes which support this feature and want to use the encryption?
Expose a configuration in config-map to control this.

Reply:
> As this feature is backed by feature-gate if someone is still on the older kubernetes which support this feature and want to use the encryption? Expose a configuration in config-map to control this.

Ceph-CSI is only tested for k8s v1.21+, I doubt we should do that.

Reply:
I am not confident if beta/alpha versions work the same as stable. 

Let's just only support the stable version.

Reply:
>Ceph-CSI is only tested for k8s v1.21+, I doubt we should do that.

Am fine, but it takes out the possibility of  using this feature with beta kubernetes (which is mostly equal to stable)

> I am not confident if beta/alpha versions work the same as stable.

As it's a template change I think if the deployment is passed and basic testing that should be enough.

Reply:
@leseb @travisn any comments on this one?

Reply:
Do we have any precedent on exposing features only available through feature gates?

Reply:
> Do we have any precedent on exposing features only available through feature gates?

I don't think so,
 k8s v1.20 for which this feature is stable is out since more than a year,
I don't think we need to have such an option just to enable support on very old k8s versions without testing or any explicit need for it.(just adds code and documentation which may never be used).

We can add it later if there's any request/need for it.